sklearn/tree/_utils.pyx

```diff
@@ -458,3 +84,212 @@ def _any_isnan_axis0(const float32_t[:, :] X):
                     isnan_out[j] = True
                     break
     return np.asarray(isnan_out)
+
+
+# =============================================================================
+# WeightedFenwickTree data structure
+# =============================================================================
+
```

Comment:
This does not add much to the `cdef class` declarations that are explicit enough.

```suggestion
```